plugins/connection/aws_ssm.py

```diff
@@ -301,10 +306,10 @@ def start_session(self):
             raise AnsibleError("failed to find the executable specified %s."
                                " Please verify if the executable exists and re-try." % executable)
 
-        profile_name = ''
+        profile_name = self.get_option('profile')
         region_name = self.get_option('region')
         ssm_parameters = dict()
-        client = self._get_boto_client('ssm', region_name=region_name)
+        client = self._get_boto_client('ssm', region_name=region_name, profile_name=profile_name)
```

Comment:
There are also a couple of calls to `self._get_boto_client('s3')` when using s3 for file transfers, these will need updating.

Reply:
I've passed `profile_name` to the other calls, I wasn't sure what the best option here would be - I'm passing it thru `_get_url`, because I'm unclear on what the preferred conventions are. Happy to change to do this differently of course.

Sorry for taking so long!